sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
Thanks for confirming @ragv. Let's try to make the common tests use the deterministic dense solver then.

Note: `np.array([[0,]*5]*5, dtype=float)` is better written `np.zeros(shape=(5, 5), dtype=np.float64)`.